packages/react/src/components/ActionList/ActionListSeparator.tsx

```diff
@@ -0,0 +1,30 @@
+import React, { forwardRef } from 'react';
+import classnames from 'classnames';
+import { useActionListContext } from './ActionListContext';
+
+type ActionListSeparatorProps = React.HTMLAttributes<HTMLLIElement>;
+
+const ActionListSeparator = forwardRef<HTMLLIElement, ActionListSeparatorProps>(
+  ({ className, ...props }, ref) => {
+    const { role: contextRole } = useActionListContext();
+    // list and listbox roles only support listitem or option roles respectively
+    // see https://github.com/w3c/aria/issues/1889
+    const separatorRole = ['list', 'listbox'].includes(contextRole as string)
+      ? 'presentation'
+      : 'separator';
```

Comment:
To avoid casting - 

```suggestion
    const separatorRole = contextRole && ['list', 'listbox'].includes(contextRole)
      ? 'presentation'
      : 'separator';
```

Reply:
In this case, I don't think this really saves us a whole lot. `[string].includes(null)` should absolutely be a valid check.  If we're worried about the cast, we could throw a `@ts-ignore` but I think that's less desirable.